src/module/item/ability/helpers.ts

```diff
@@ -40,4 +42,39 @@ function addSheetFrequencyListeners(item: ItemPF2e & SourceWithFrequencyData, ht
     });
 }
 
-export { addSheetFrequencyListeners, normalizeActionChangeData };
+/** Create data for the "self-applied effect" drop zone on an ability or feat sheet. */
+function createSelfEffectSheetData(
+    data: SelfEffectReference | null
+): SelfEffectReference | { name: string; empty: true } {
+    if (data && !data.img) {
+        type MaybeIndexData = ((ClientDocument | CompendiumIndexData) & { img?: unknown }) | null;
+        const indexEntry: MaybeIndexData = fromUuidSync(data.uuid);
```

Comment:
Since sheet data is async, couldn't you use the async version of this function?

Reply:
The one that makes a server call, in order to get the same data already indexed by default?